Test/Unit/Model/Import/RuleModelTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Taxjar\SalesTax\Test\Unit\Model\Import;
+
+use Magento\Framework\Api\AttributeValueFactory;
+use Magento\Framework\Api\ExtensionAttributesFactory;
+use Magento\Framework\Data\Collection\AbstractDb;
+use Magento\Framework\Event\ManagerInterface;
+use Magento\Framework\Model\Context;
+use Magento\Framework\Model\ResourceModel\AbstractResource;
+use Magento\Framework\Registry;
+use Magento\Tax\Model\Calculation;
+use Magento\Tax\Model\Calculation\Rule\Validator;
+use Magento\Tax\Model\ClassModel;
+use Taxjar\SalesTax\Model\Import\RuleModel;
+use Taxjar\SalesTax\Test\Unit\UnitTestCase;
+
+class RuleModelTest extends UnitTestCase
+{
+    public function testAfterSave()
+    {
+        $mockEventManager = $this->createMock(ManagerInterface::class);
+        $mockEventManager->expects($this->exactly(4))
+            ->method('dispatch')
+            ->withConsecutive(
+                ['model_save_after'],
+                ['clean_cache_by_tags'],
+                ['tax_rule_save_after'],
+                ['tax_settings_change_after']
+            );
+
+        $mockContext = $this->createMock(Context::class);
+        $mockContext->expects($this->once())->method('getEventDispatcher')->willReturn($mockEventManager);
+
+        $sut = $this->getMockBuilder(RuleModel::class)
```

Comment:
Do you need to mock the SUT here. Would it work to just instantiate the actual object with mock dependencies and then call the afterSave method?

Reply:
I don't really have to mock it here, and it would work by instantiating the object and passing the mock dependencies thru the constructor. The issue that I had with that is in the Tax module's Rule class constructor here:
![image](https://user-images.githubusercontent.com/47947793/134700801-2765d238-a690-4d8d-b5a2-873905975840.png)

The `_init` method in the tax module's class that our RateModel class extends from sets and retrieves the resource class, so we end up mocking deps that could be avoided by bypassing the constructor, but I can refactor this if you'd prefer.